stm32f4-rtic-smart-keyboard/src/bin/minif4_36-rev2021_4-lhs.rs

```diff
@@ -0,0 +1,229 @@
+#![no_main]
+#![no_std]
+
+#[cfg(not(custom_board))]
+mod board {
+    use stm32f4xx_hal as hal;
+
+    pub const COLS: usize = 5;
+    pub const ROWS: usize = 4;
+
+    #[rustfmt::skip]
+    pub const KEYMAP_INDICES: [[Option<u16>; COLS]; ROWS] = [
+        [ Some(0),  Some(1),  Some(2),  Some(3),  Some(4)],
+        [Some(10), Some(11), Some(12), Some(13), Some(14)],
+        [Some(20), Some(21), Some(22), Some(23), Some(24)],
+        [Some(30), Some(31), Some(32),     None,     None],
+    ];
+
+    pub use stm32f4_rtic_smart_keyboard::app_prelude::VID;
+
+    pub const PID: u16 = 0x0005;
+    pub const MANUFACTURER: &str = "smart-keyboard";
+    pub const PRODUCT: &str = "MiniF4-36 rev2021.4 LHS";
+
+    pub type Keyboard = usbd_smart_keyboard::input::Keyboard<COLS, ROWS, Matrix>;
+
+    pub type PressedKeys = usbd_smart_keyboard::input::PressedKeys<COLS, ROWS>;
+
+    pub struct Matrix(
+        pub keyberon::matrix::DirectPinMatrix<hal::gpio::ErasedPin<hal::gpio::Input>, COLS, ROWS>,
```

Comment:
`hal::gpio::ErasedPin<hal::gpio::Input>` could probably be named "InputPin". IIRC, the rp2040_rtic code has a similar type alias.